### PR TITLE
soc: it8xxx2: optimize __soc_handle_irq and __soc_is_irq functions

### PR DESCRIPTION
**Pull request by Dino-Li**
This reduces code size slightly.

Signed-off-by: Dino Li <Dino.Li@ite.com.tw>

### CODE REVIEW

File: soc/riscv/riscv-ite/common/soc_irq.S [hunk 1]
```diff
@@ -20,17 +20,11 @@ GTEXT(__soc_handle_irq)
 /*
  * SOC-specific function to handle pending IRQ number generating the interrupt.
  * Exception number is given as parameter via register a0.
+ * Jump to get_irq() function directly and return to caller by its
+ * ret instruction.
  */
 SECTION_FUNC(exception.other, __soc_handle_irq)
-	/* store ra */
-	addi sp, sp, -4
-	sw ra, 0(sp)
```
Comment (fkokosinski) [inline soc/riscv/riscv-ite/common/soc_irq.S:27]: Why are we no longer storing `ra` here? `ra` is caller-saved in RISC-V.

File: soc/riscv/riscv-ite/common/soc_irq.S [hunk 2]
```diff
@@ -20,17 +20,11 @@ GTEXT(__soc_handle_irq)
...
-	/* get interrupt number and clear interrupt status of soc */
-	call get_irq
-	/* restore ra */
-	lw ra, 0(sp)
-	addi sp, sp, 4
-	ret
+	j get_irq
```
Comment (fkokosinski) [inline soc/riscv/riscv-ite/common/soc_irq.S:27]: Is this change really necessary? How much space does it save? We shouldn't really use `j` for function calls here.

Comment (edersondisouza) [reply soc/riscv/riscv-ite/common/soc_irq.S:27]: I wonder if there's any chance that `get_irq` ends up too far to be called via `j` - in which case you'd need `jr` (or abuse `tail`).

Comment (Dino-Li) [reply soc/riscv/riscv-ite/common/soc_irq.S:27]: It should be safe to use `j` here, because the maximum flash size of it8xxx2 series is 1MB.

**Review by edersondisouza — APPROVED** [review]

File: soc/riscv/riscv-ite/common/soc_irq.S (hunk 2 above)
Comment (fkokosinski) [reply soc/riscv/riscv-ite/common/soc_irq.S:27]: Yes, `j` is indeed safe to jump within the 1MB range (section 2.5. of the [RISC-V spec](https://github.com/riscv/riscv-isa-manual/releases/download/draft-20220604-4a01cbb/riscv-spec.pdf)). Thanks answering my question.

**Review by fkokosinski — APPROVED** [review]
